[IoT State Channels] Upgrade to go-perun v0.9.0

manoranjith opened the pull request:
<!-- Provide a general summary of your changes in the title above

Please read our contribution guidelines and sign the Contributor License
Agreement (CLA) before submitting the pull request. Also, check if there are no
other open pull requests targeting the same issue. -->

#### Description
<!-- Describe your changes in detail. -->

Implements #249.

##### Category
<!-- Tell us what type of issue does your pull request target.
You can uncomment one of the following options: -->

<!-- Bug Fix -->
<!-- Improvement -->
Implementation Task

##### Relevant issue
<!-- Provide a link to the related issue. You can use the following keywords
and the issue number: "fixes", "resolves", "relates to". E.g.: closes #21

We accept only pull requests related to open issues. If you're suggesting a new
feature, improvement or fixing a bug that is not yet reported, please discuss it in
an issue before submitting a pull request. -->

#249
 
#### Testing
<!-- Tell us how you have tested the changes. -->

All existing tests should run.

##### Steps to run the tests
<!-- Describe a set of steps to run the tests relevant to this change. -->

1. 
2. 
3.  

#### Checklist 
<!-- Please check if the pull request fulfils these requirements: -->

- [x] Name is added to the NOTICE file, if it is not present already.
- [x] Changes are rebased onto the target branch.

Commit:
Update go version to 1.19 in go.mod, circleci

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Update go-perun to v0.9.0 in go.(mod|sum)

- Also update the path for perun generic library packages to from
  "perun.network/go-perun/pkg/sync" to "polycry.pt/poly-go/sync" as
  this library was moved in the newer version of go-perun.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Adapt according to update go-perun v0.9.0

- Use LatestSignerForChainID instead of NewEIP155Signer.

- Use channel.Asset type instead of wallet.Address in relevant places.

- Use Update instead of UpdateBy with new signature for updater func.

- Rename wallet.Address.(Equals -> Equal)

- Update all mocks with latest version of mockery (separate this
  change).

- Use updated signature for NewBus and pass the serializer explicitly.

- Use updated signature for listner.Accept and pass the serializer
  explicitly.

- Replace usage of Bytes method on Address with MarshalBinary.

- Adapt to rename of some of the data types.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Update testdata for persistence tests

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Upgrade golangci-lint to latest (1.53.3)

- Because, the older version is not compatible with go version 1.19.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Fix linter error - gofumpt

- Because the rules for newer version of gofumpt has been updated.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Update linter config - remove depricated linters

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Fix linter config - depgaurd

- Because the syntax for configuring this linter has changed.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Fix linter error - doc for main pkg

- Because the updated version of linter requires doc comments for main
  packages as well.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Fix linter error - replace usage of ioutil pkg

- Because package has been depricated since go 1.17.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Fix linter error - unused

- These errors were not detected by older version of revive.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Update proto and grpc compilers to latest versions

- Also, update the syntax for protoc command, as the compiler for grpc
  bindings has now been separated from the one for protobuf bindings.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Fix linter error - nolint

- go1.19 introduced new formatting directives have been for comments.

- The new syntax is "//toolname:directive".

- For more details, see the syntax section in this page -
  "https://go.dev/doc/comment".

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
Fix linter error - goimports

- Run goimports with "-local github.com/hyperledger-labs/perun-node"
  option.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

danielksan81 reviewed (dismissed):
Changes works as well locally.

LGTM

Commit:
Update go version, test command in Readme

- For tests using ganache-cli, parallelism should be set to 1 to ensure
  the correct incrementation of nonce used in creating transactions.

Signed-off-by: manoranjith <ponraj.manoranjitha@in.bosch.com>

manoranjith commented:
Updated the Readme, as discussed. Could you add your review again ? @danielksan81

danielksan81 reviewed (approved):
Good. Covers now also changes for go version in Readme.
